cms/djangoapps/contentstore/utils.py

```diff
@@ -160,7 +160,10 @@ def get_lms_link_for_certificate_web_view(user_id, course_key, mode):
 
 def course_image_url(course):
     """Returns the image url for the course."""
-    loc = StaticContent.compute_location(course.location.course_key, course.course_image)
+    try:
+        loc = StaticContent.compute_location(course.location.course_key, course.course_image)
+    except InvalidKeyError:
+        return ''
     path = StaticContent.serialize_asset_key_with_slash(loc)
     return path
```

Comment:
Yeah, that's what I'd do.

-Cale
On Jul 2, 2015 6:13 AM, "Awais Jibran" <notifications@github.com> wrote:

> In cms/djangoapps/contentstore/utils.py
> <https://github.com/edx/edx-platform/pull/8499#discussion_r33764169>:
>
> > @@ -160,7 +160,10 @@ def get_lms_link_for_certificate_web_view(user_id, course_key, mode):
> >
> >  def course_image_url(course):
> >      """Returns the image url for the course."""
> > -    loc = StaticContent.compute_location(course.location.course_key, course.course_image)
> > +    try:
> > +        loc = StaticContent.compute_location(course.location.course_key, course.course_image)
> > +    except InvalidKeyError:
> > +        return ''
> >      path = StaticContent.serialize_asset_key_with_slash(loc)
> >      return path
>
> @cpennington <https://github.com/cpennington> So we can do something like
> this?
>
> try:
>      loc = StaticContent.compute_location(course.location.course_key, course.course_image)
> except InvalidKeyError:
>        return None
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/8499/files#r33764169>.
>